make-dist

```diff
@@ -85,13 +85,33 @@ _python_autoselect() {
 build_dashboard_frontend() {
   CURR_DIR=`pwd`
   TEMP_DIR=`mktemp -d`
+
   $CURR_DIR/src/tools/setup-virtualenv.sh --python=$(_python_autoselect) $TEMP_DIR
   $TEMP_DIR/bin/pip install nodeenv
   $TEMP_DIR/bin/nodeenv -p --node=10.16.0
   cd src/pybind/mgr/dashboard/frontend
+
+  DEFAULT_LANG=`cat package.json | jq -r .config.locale`
+  if [ -z "$DASHBOARD_FRONTEND_LANGS" ]; then
+    BUILD_TARGET=":${DEFAULT_LANG}"
+  else
+    if [ "$DASHBOARD_FRONTEND_LANGS" == "ALL" ]; then
+      BUILD_TARGET=":*"
+    else
+      DASHBOARD_FRONTEND_LANGS=`echo "$DASHBOARD_FRONTEND_LANGS" | sed 's/ /,/g'`
+      if [[ $DASHBOARD_FRONTEND_LANGS != *"${DEFAULT_LANG}"* ]]; then
+        # default language must be always built
+        DASHBOARD_FRONTEND_LANGS="${DASHBOARD_FRONTEND_LANGS},${DEFAULT_LANG}"
```

Comment:
Why has default language to be built? Let's say I just want a Japanese-only front-end, should I need to build the default too?

Reply:
The current logic in the backend and frontend is to fallback to a "default" language if it can't match the browser request. Since currently the default language is static "it's always en-US" we must ensure that en-US is built.

To support the use case you're mentioning, we need to support a dynamic default. For instance, if the user only builts japanese, then japanese will be the default. The problem here is that if the user builds two languages we also need to know which one to use as the default. And that requires yet another thing for the user to configure, which IMO is not worth the effort.

Reply:
Thanks for the clarification!